tests/core/matrix.rs

```diff
@@ -970,6 +970,14 @@ mod inversion_tests {
             }
         }
 
+        #[test]
+        fn test_inversion_failure_leaves_matrix_unchanged(m in matrix4()) {
+                let original_matrix = m.clone();
+                if m.try_inverse().is_none() {
```

Comment:
Does this branch actually get hit? I'm not sure how often proptest generates non-invertible matrices. Would a hardcoded case work better?

Reply:
Agree with @Ralith, I think the likelihood of `matrix4()` producing singular matrices is likely (very) small (though I haven't checked), so a test with a fixed input is better in this case.